services/feedz/feedz.service.js

```diff
@@ -80,6 +93,25 @@ class FeedzVersionService extends BaseJsonService {
     })
   }
 
+  async fetchItems({ json }) {
+    if (json.items.length > 0 && json.items.every(i => !i.catalogEntry)) {
+      const items = await Promise.all(
+        json.items.map(i =>
+          this._requestJson({
+            schema: singlePageSchema,
+            url: i['@id'],
+            errorMessages: {
+              404: 'repository or package not found',
+            },
+          })
+        )
+      )
+      return { items }
+    } else {
+      return json
+    }
```

Comment:
Bit of a nit since it probably won't have much of a runtime impact, but could we invert the conditional with an early return on the iterator to hit the return path?

E.g. would something like this work? (hand typed in browser so might be botched but hopefully still conveys the thought)

```suggestion
    if (json.items.length === 0 || json.items.any(i => i.catalogEntry)) {
      return json
    } else {
      const items = await Promise.all(
        json.items.map(i =>
          this._requestJson({
            schema: singlePageSchema,
            url: i['@id'],
            errorMessages: {
              404: 'repository or package not found',
            },
          })
        )
      )
      return { items }
    }
```

Reply:
Yeah, that should make the "no need to download anything" path a little bit faster.